src/registry/currency/edit.tsx

```diff
@@ -159,18 +159,8 @@ const DefaultValue: React.FC = () => {
   );
 };
 
-const ComparisonValueInput: React.FC = () => {
-  return (
-    <NumberField
-      name="conditional.eq"
-      label={
-        <FormattedMessage
-          description="Component property 'conditional.eq' label"
-          defaultMessage="Has the value"
-        />
-      }
-    />
-  );
+const renderComparisonValueInput: React.FC<ComparisonValueProps> = props => {
```

Comment:
what about:

```tsx
const ComparisonValueInput = NumberField;
```

All the `ComparisonValueProps` are/should be assignable to `NumberField`, so I'd expect this to type-check.

If it doesn't, a higher order component could work:

```tsx
const ComparisonValueInput = asComparisonValue(NumberField);
```